firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigContainer.java

```diff
@@ -189,6 +193,38 @@ public boolean equals(Object o) {
     return containerJson.toString().equals(that.toString());
   }
 
+  /** Creates a map where the key is the config key and the value if the experiment description. */
+  private Map<String, JSONObject> createExperimentsMap(JSONArray allExperiments)
+      throws JSONException {
+    Map<String, JSONObject> experimentsMap = new HashMap<>();
+    if (allExperiments == null) {
+      return experimentsMap;
+    }
+
+    // Iterate through all experiments to check if it has the `affectedParameterKeys` field.
+    for (int i = 0; i < allExperiments.length(); i++) {
+      JSONObject experiment = allExperiments.getJSONObject(i);
+      if (!experiment.has(AFFECTED_PARAMETER_KEYS)
+          || experiment.getString(EXPERIMENT_ID).startsWith(ROLLOUT_ID_PREFIX)) {
+        continue;
+      }
+
+      // Since a config key can only have one experiment associated with it, map the key to the
+      // experiment.
+      JSONArray affectedKeys = experiment.getJSONArray(AFFECTED_PARAMETER_KEYS);
+      for (int j = 0; j < affectedKeys.length(); j++) {
+        String key = affectedKeys.getString(j);
+        JSONObject experimentsCopy = new JSONObject(experiment.toString());
+        // Removing `affectedParameterKeys` because its values never come in the same order which
```

Comment:
thinking: and because we're diffing the metadata w.r.t. the key(s) itself so we don't need to diff the key

Reply:
yeah that makes sense, removed this